tyssue/core/sheet.py

```diff
@@ -91,6 +91,102 @@ def get_neighborhood(self, face, order):
 
         return neighbors.reset_index(drop=True).loc[1:]
 
+    def sheet_extract(self, key_word_column, coords=['x', 'y', 'z']):
+        """ Extract a new sheet from the embryo sheet
+        that correspond to a key word that define a face.
+        Parameters
+        ----------
+        sheet: a :class:Sheet object
+        coords
+        Returns
+        -------
+        sheet_fold_patch_extract :
+            subsheet corresponding to the fold patch area.
+
+        """
+        x, y, z = coords
+        datasets = {}
+
+        datasets['face'] = self.face_df[
+            self.face_df[key_word_column] == True].copy()
+        datasets['edge'] = self.edge_df[self.edge_df['face'].isin(
+            datasets['face'].index)].copy()
+        datasets['vert'] = self.vert_df[self.vert_df['srce_o'].isin(
```

Comment:
what is the `srce_o` column? You can get the relevant vertices with `self.edge_df['srce'].unique()` as in
```python
datasets['vert'] = self.vert_df.loc[self.edge_df['srce'].unique()]
```